src/fixed.jl

```diff
@@ -7,6 +7,9 @@ struct Fixed{T <: Signed,f} <: FixedPoint{T,  f}
     Fixed{T, f}(i::Integer, _) where {T,f} = new{T, f}(i % T)
     Fixed{T, f}(x) where {T,f} = convert(Fixed{T,f}, x)
     Fixed{T, f}(x::Fixed{T,f}) where {T,f} = x
+    Fixed{T, f}(x::Char) where {T,f} = throw(ArgumentError("Fixed cannot be constructed from a Char"))
+    Fixed{T, f}(x::Complex) where {T,f} = Fixed{T, f}(real(x))
```

Comment:
`real` returns the real part even if there is a non-zero complex part. Would it be better to use `convert` and get the `InexactError`?

Reply:
Yes, that is a blunder. The reason I ended here is that `Complex{S}`, for some reason, didn't match the ambiguous signature. Guess it will have to be `Fixed{T, f}(convert(real(eltype(x)), x))`. Will push a fix.